examples/prom/nonlinear_elasticity_global_rom.cpp

```diff
@@ -1691,16 +1912,812 @@ void RomOperator::Mult_FullOrder(const Vector& vx, Vector& dvx_dt) const
+
+    // Compute G of size (NB * nsnap) x NQ, but only store its transpose Gt.
+    CAROM::Matrix Gt(NQ, NB * nsnap, true);
+    int current_size = 0;
+    int i_start = 0;
+    int i_end = 0;
+    int outer_loop_length = 0;
+    if (!window_ids)
+    {
+        current_size = nsnap;
+        outer_loop_length = 1;
+        i_end = nsnap;
+    }
+    else
+    {
+        outer_loop_length = window_ids->dim() - 1;
+    }
+
+    // For 0 <= j < NB, 0 <= i < nsnap, 0 <= e < ne, 0 <= m < nqe,
+    // G(j + (i*NB), (e*nqe) + m)
+    // is the coefficient of v_j^T M(p_i) V v_i at point m of element e,
+    // with respect to the integration rule weight at that point,
+    // where the "exact" quadrature solution is ir0->GetWeights().
+
+    Vector x_i(BX_snapshots->numRows());
+    Vector v_j(BV->numRows());
+
+    Vector r(nqe);
+
+    int skip = 0;
+    const int nsnapPerSet = nsnap / nsets;
+    if (skipFirstW)
+    {
+        MFEM_VERIFY(nsets * nsnapPerSet == nsnap, "");
+        skip = 1;
+    }
+
+    // Get prolongation matrix
+    const Operator *P = fespace_X->GetProlongationMatrix();
+    if (!P)
+    {
+        MFEM_ABORT("P is null, generalize to serial case")
+    }
+
+    // Outer loop for time windowing
+    for (int oi = 0; oi < outer_loop_length; ++oi)
+    {
+        if (window_ids)
+        {
+            i_start = window_ids->item(oi) - 1;
+            i_end = window_ids->item(oi + 1) - 1;
+            current_size = i_end - i_start;
+            cout << "i_start = " << i_start << endl;
+            cout << "Number of NNLS equations: " << NB * current_size << endl;
+            Gt.setSize(NQ, NB * current_size);
+        }
+
+        // For every snapshot in batch
+        for (int i = i_start; i < i_end; ++i)
+        {
+            // Set the sampled dofs from the snapshot matrix
+            for (int j = 0; j < BX_snapshots->numRows(); ++j)
+                x_i[j] = (*BX_snapshots)(j, i) + x0[j];
+
+            // Get prolongated dofs
+            Vector px_i;
+            Vector elfun;
+
+            px_i.SetSize(P->Height());
+            P->Mult(x_i, px_i);
+
+            if (skipFirstW && i > 0 && i % nsnapPerSet == 0)
+                skip++;
+
+            // TODO: is it better to make the element loop the outer loop?
+            // For each element
+            for (int e = 0; e < ne; ++e)
+            {
+                // Get element and its dofs and transformation.
+                Array<int> vdofs;
+                DofTransformation *doftrans = fespace_X->GetElementVDofs(e, vdofs);
+                const FiniteElement &fe = *fespace_X->GetFE(e);
+                ElementTransformation *eltrans = fespace_X->GetElementTransformation(e);
+                px_i.GetSubVector(vdofs, elfun);
+
+                if (doftrans)
+                {
+                    MFEM_ABORT("Doftrans is true, make corresponding edits")
+                }
+
+                // For each basis vector
```

Comment:
The same basis is used for all time windows, right? Another possibility is to construct a smaller basis for each window, which should reduce the size of the EQP system, for faster setup without significant loss of accuracy. If this is not done in this PR, it would be helpful at least to comment about this.

Reply:
Yes, it's the same basis. I agree! I stopped working on the timewindowing after we decided not to use it for the paper and I'll likely not put it in this PR. But I'll add a comment and a feature request issue so that someone can pick it up